### ISSUE TITLE
Allow report-to in CSP and CSPRO meta tags

### ISSUE THREAD

**Issue by ScottHelme** [w3c/webappsec-csp#348]
We talked about allowing CSPRO in meta tags in #277

We also talked about why `report-uri` isn't allowed in meta in #278

Given the above it seems like allowing `report-to` in both CSP and CSPRO meta tags would be safe and convenient. No special handling is required for reports in JS and we can use the [Reporting API](https://www.w3.org/TR/reporting/) implementation in the browser. The 2nd issue shouldn't be a problem because alongside the HTML injection for the meta tag the attacker would need a header injection flaw to get the `Report-To` header in, at which point they could just set their own CSP anyway.

**Comment by dveditz** [comment]
Right, so why didn't the entity setting the report-to header set a CSP header which uses it? What scenario would this ability help?

Doesn't (currently) look like a disclosure risk--unlike report-uri--because the reporting endpoint is still defined in headers. That's good. It does mean that if a site defines a report-to for some reasons then an injected meta CSP could try to overwhelm it. The site might be defining report-to for use with their legit non-meta CSP header or for some other purpose. There might unintended data disclosure if the report-to was meant for low-integrity data (perhaps using a 3rd party service) and then the injected meta-CSP had a restrictive policy that reported violations that include PII (maybe medical diagnostic codes?).

I'm also worried that as some point someone will argue that the report-to header itself should be able to be defined via a <meta> tag, and since that's in a separate specification no one will notice that the combination of that feature request and this feature request results in the situation we rejected when we banned CSPRO from being used in a <meta> tag.

**Comment by eligrey** [comment]
This would be very useful for JS-handled CSP violations as used as a client-side layer of analytics auditing.
